plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java

```diff
@@ -172,6 +175,10 @@ private void doVacuum(
         accessControl.checkCanDeleteFromTable(null, tableName);
 
         TableSnapshot tableSnapshot = transactionLogAccess.loadSnapshot(tableName, handle.getLocation(), session);
+        ProtocolEntry protocolEntry = transactionLogAccess.getProtocolEntry(session, tableSnapshot);
+        if (protocolEntry.getMinWriterVersion() > MAX_WRITER_VERSION) {
```

Comment:
Should we be checking write features as well?

Reply:
Yes, but we have to introduce unreachable code because the current max writer version is 5 and writer features exist since 7. 

Added TODO comment with link to https://github.com/trinodb/trino/issues/15873